### ISSUE TITLE
ReducerStore addon

### ISSUE THREAD

**Issue by mridgway** [yahoo/fluxible#220]
This won't be part of core, but it should be easy to create a shared store that calls into reducers. This would most likely be a separate npm package or part of `fluxible-addons-react`.

Here's what I was thinking:

```js
createReducerStore(storeName, {
    'RECEIVED_MESSAGE': (state = 0, payload) => ({
        return state + 1;
    })
});
```

This would allow a reducer to be hooked up to the `RECEIVED_MESSAGE` event. The `storeName` would allow you to have multiple stores that you could then register the Fluxible instance just like normal.

**Comment by cesarandreu** [comment]
+∞

Would love to have this to play around with it.

**Comment by mridgway** [comment]
I've been playing around with the following implementation: https://gist.github.com/mridgway/2924b9c71e7d8d925e55

What I've found is that the whole hot-loading thing is dependent on your application being loaded from a react component since react-hot-loader only works with... well... react.

So there needs to be some additional hot loading magic for applications that bootstrap from a client.js file rather than some App.jsx.

**Comment by geekyme** [comment]
what would be the impact of this?

**Comment by mridgway** [comment]
Nothing really. You'd have the option of using a ReducerStore which is just a subclass of BaseStore. In theory this would make it easier use hot reloading, but I have not been able to get it to work successfully yet.

**Comment by mridgway** [comment]
This will probably be released as a separate package instead of as part of Fluxible since it may require some rework before becoming stable.

**Comment by geekyme** [comment]
:+1: I agree. By the way just to clarify, would I be able to use a single store approach and hook on multiple reducers? or does this simply extend the multi-store approach to make each store use the reducers that are declared for each of them?

**Comment by mridgway** [comment]
This was released as `fluxible-reducer-store@0.1.0`. You can find the source and documentation here: https://github.com/yahoo/fluxible/tree/master/packages/fluxible-reducer-store
